app/subscriber/src/features/event-of-the-day/EventOfTheDayPreview.tsx

```diff
@@ -0,0 +1,113 @@
+import { DateFilter } from 'components/date-filter';
+import moment from 'moment';
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { toast } from 'react-toastify';
+import { useContent, useLookup, useReports } from 'store/hooks';
+import { Col, IReportResultModel, Loading, Show } from 'tno-core';
+
+import * as styled from './styled';
+
+const EventOfTheDayPreview: React.FC = () => {
+  const [
+    {
+      eventOfTheDay: { filter: eventOfTheDayFilter },
+    },
+    { storeEventofTheDayDateFilter: storeFilter },
+  ] = useContent();
+  const [params] = useSearchParams();
+  const dateUrlParam = params.get('date')
+    ? params.get('date')
+    : moment(new Date()).endOf('day').toISOString();
+  const [, { findInstanceForReportIdAndDate }] = useReports();
+  const [{ settings }] = useLookup();
+  const [date, setDate] = React.useState<string>();
+
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [preview, setPreview] = React.useState<IReportResultModel | undefined>();
+
+  const clear = () => {
+    setIsLoading(true);
+    setPreview(undefined);
+  };
+
+  // Used to refresh preview data for newly selected date
+  const handleChangeReportDate = React.useCallback(
+    async (overrideDate?: string) => {
+      try {
+        clear();
```

Comment:
If `clear()` is only used here, there isn't much of a reason to have a separate function for it.